packages/core/src/application.ts

```diff
@@ -280,141 +32,89 @@ export class Application extends Context {
-   * app.route(route);
+   * app.servers([
+   *  new RestServer(app, options),
+   *  new GRPCServer(app, options),
+   * ]);
+   * // Creates a binding for "servers.RestServer" and a binding for
+   * // "servers.GRPCServer";
    * ```
    *
-   * @param route The route to add.
+   * @param {Array<Server>} servers An array of instantiated Server instances.
+   * @returns {Binding}
+   * @memberof Application
    */
-  route(route: RouteEntry): Binding;
-
-  route(
-    routeOrVerb: RouteEntry | string,
-    path?: string,
-    spec?: OperationObject,
-    controller?: ControllerClass,
-    methodName?: string,
-  ): Binding {
-    if (typeof routeOrVerb === 'object') {
-      const r = routeOrVerb;
-      return this.bind(`routes.${r.verb} ${r.path}`).to(r);
-    }
-
-    if (!path) {
-     throw new AssertionError({
-       message: 'path is required for a controller-based route',
-     });
-    }
-
-    if (!spec) {
-     throw new AssertionError({
-       message: 'spec is required for a controller-based route',
-     });
-    }
-
-    if (!controller) {
-     throw new AssertionError({
-       message: 'controller is required for a controller-based route',
-      });
-    }
-
-    if (!methodName) {
-      throw new AssertionError({
-        message: 'methodName is required for a controller-based route',
-      });
-    }
-
-    return this.route(new ControllerRoute(
-      routeOrVerb,
-      path,
-      spec,
-      controller,
-      methodName));
-  }
-
-  api(spec: OpenApiSpec): Binding {
-    return this.bind(CoreBindings.API_SPEC).to(spec);
+  public servers(servers: Server[]): Binding[] {
+    const self = this;
+    return servers.map((server) => {
+      return self.server(server);
+    });
   }
 
   /**
-   * Get the OpenAPI specification describing the REST API provided by
-   * this application.
+   * Start the application, and all of its registered servers.
    *
-   * This method merges operations (HTTP endpoints) from the following sources:
-   *  - `app.api(spec)`
-   *  - `app.controller(MyController)`
-   *  - `app.route(route)`
-   *  - `app.route('get', '/greet', operationSpec, MyController, 'greet')`
+   * @returns {Promise}
+   * @memberof Application
    */
-  getApiSpec(): OpenApiSpec {
-    const spec = this.getSync(CoreBindings.API_SPEC);
-
-    // Apply deep clone to prevent getApiSpec() callers from
-    // accidentally modifying our internal routing data
-    spec.paths = cloneDeep(this.httpHandler.describeApiPaths());
-
-    return spec;
-  }
-
-  protected _logError(
-    err: Error,
-    statusCode: number,
-    req: ServerRequest,
-  ): void {
-    console.error(
-      'Unhandled error in %s %s: %s %s',
-      req.method,
-      req.url,
-      statusCode,
-      err.stack || err,
-    );
+  public async start() : Promise<void> {
+    const self = this;
+    const promises = this._servers.map(async (server) => {
+      const inst = await self.get(server) as Server;
+      return inst.start();
```

Comment:
Perhaps `return await inst.start()`? I am never sure which form is better, I'd expect the `await` form to produce better stack traces pointing back to this anonymous lambda function.